### PR TITLE
Modified bucket_table() to compute WoE consistent with category_encoder

### PR DESCRIPTION
**Pull request by dlaprins**
#87 
Hard-coded WoE calculation in bucket_table to be consistent with the category_encoder  (v2.5.1) implementation of WoE. 

A direct call on the WOEEncoder object might be preferable, but seems inconvenient given the currrent code setup.

### CODE REVIEW

**Comment by anilkumarpanda** [comment]
Hey @dlaprins thanks for the PR. Can you write an unit test to test you change ?

File: skorecard/reporting/report.py [hunk 1]
```diff
@@ -17,7 +17,7 @@ def build_bucket_table(
     column: str,
     bucketer=None,
     bucket_mapping: Optional[BucketMapping] = None,
-    epsilon=0.00001,
+    epsilon=1,
```
Comment (sbjelogr) [inline skorecard/reporting/report.py:20]: Note that the numerator and denominator of the WoE formula are decimals: changing epsilon to 1 will mess up heavily the WoE calculation. The very small number is there to ensure that there is no division by zero.

File: skorecard/reporting/report.py [hunk 2]
```diff
@@ -123,7 +123,10 @@ def build_bucket_table(
     stats["% Event"] = stats["Event"] / stats["Event"].sum()
     stats["% Non-event"] = stats["Non-event"] / stats["Non-event"].sum()
 
-    stats["WoE"] = ((stats["% Non-event"] + epsilon) / (stats["% Event"] + epsilon)).apply(lambda x: np.log(x))
+    event_percentage = (stats["Event"] + epsilon) / (stats["Event"].sum() + 2 * epsilon)
+    non_event_percentage = (stats["Non-event"] + epsilon) / (stats["Non-event"].sum() + 2 * epsilon)
+    stats["WoE"] = (event_percentage / non_event_percentage).apply(lambda x: np.log(x))
```
Comment (sbjelogr) [inline skorecard/reporting/report.py:128]: The WoE is normally calculated in the following function

```python
from skorecard.metrics.metrics import woe_1d
```

and then this is propagated through calculations (see the `WoEEncoder`). 
This should guarantee consistency across the package. I would suggest using it in here

**Review by sbjelogr — COMMENTED** [review]
Thanks or this PR. Please have a look at the comments in this review.

File: skorecard/reporting/report.py (hunk 2 above)
Comment (dlaprins) [reply skorecard/reporting/report.py:128]: Hi Sandro, thank you very much for your comments, I am glad to see you are still looking after the project.
 
I was under the impression that WoEEncoder was deprecated (line 18 of _WoEencoder.py) in favor of category_encoders WOEEncoder , which is the one in use in the Skorecard class (line 216 of skorecard.py).
This is also why I set epsilon=1: I agree with you that it makes a lot more sense to have it small, but skorecard.py uses the default value for category_encoders' WOEEncoder, namely "regularisation=1" (line 77 of  
[https://github.com/scikit-learn-contrib/category_encoders/blob/master/category_encoders/woe.py](url)
).  

Would it be your suggestion to continue working with category_encoder's WOEEncoder, or with skorecard's WoEEncoder? If continuing with category_encoders' WOEEncoder, a remedy for the regularisation would be to set it back to 10**(-5) in bucket_table(), but then to also alter the call to WOEEncoder in the skorecard.py.

Comment (sbjelogr) [reply skorecard/reporting/report.py:128]: Hi @dlaprins, you are correct. Indeed the epsilon is added at denominator and numerator level, not in the ratio.
And after checking a bit more, I noticed that I missed this deprecation of WoEEncoder that happened in the issue #53.

As this method is being deprecated, it makes sense to keep the change you suggest in this PR.

File: skorecard/reporting/report.py (hunk 1 above)
Comment (sbjelogr) [reply skorecard/reporting/report.py:20]: As pointed in the comment below, epsilon=1 makes the calculations compatible with the WoEEncoder transformer.

**Pull request merged**